### PR TITLE
Add gluster RWO provisioner.

### PR DESCRIPTION
**Pull request by humblec**
Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

### CODE REVIEW

**Comment by humblec** [comment]
@wongma7 PTAL.

File: glusterblock/glusterblock-provisioner.go [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+		}
+		secret = v
+	}
+	// If not found, the last secret in the map wins as done before
+	return secret, nil
+}
+
+
+// GetSecretForPV locates secret by name and namespace, verifies the secret type, and returns secret map
+func GetSecretForPV(secretNamespace, secretName, volumePluginName string, kubeClient clientset.Interface) (map[string]string, error) {
```
Comment (wongma7) [inline glusterblock/glusterblock-provisioner.go]: Should clientset.Interface be kubernetes.Interface? to use the client-go import "k8s.io/client-go/kubernetes"

File: glusterblock/glusterblock-provisioner.go [hunk 2]
```diff
@@ -0,0 +1,355 @@
...
+        secret := make(map[string]string)
+        if kubeClient == nil {
+                return secret, fmt.Errorf("Cannot get kube client")
+        }
+        secrets, err := kubeClient.Core().Secrets(secretNamespace).Get(secretName, metav1.GetOptions{})
```
Comment (wongma7) [inline glusterblock/glusterblock-provisioner.go]: it says metav1 doesn't exist, probably it's supposed to be an alias for "k8s.io/apimachinery/pkg/apis/meta/v1" like in kubernetes code?

File: glusterblock/glusterblock-provisioner.go [hunk 3]
```diff
@@ -0,0 +1,355 @@
...
+		glog.Fatalf("Error getting server version: %v", err)
+	}
+
+	// Create the provisioner: it implements the Provisioner interface expected by
+	// the controller
+	glusterBlockProvisioner := NewglusterBlockProvisioner(clientset)
+
+	// Start the provision controller which will dynamically provision glusterblock
+	// PVs
+	pc := controller.NewProvisionController(clientset, resyncPeriod, provisionerName, glusterBlockProvisioner, serverVersion.GitVersion, exponentialBackOffOnError, failedRetryThreshold)
```
Comment (wongma7) [inline glusterblock/glusterblock-provisioner.go]: the latest version of library requires 4 new arguments at the end https://github.com/kubernetes-incubator/external-storage/blob/master/nfs/cmd/nfs-provisioner/main.go#L120 

usually I just use default like this and I have to import the leaderelection package https://github.com/kubernetes-incubator/external-storage/blob/master/nfs/cmd/nfs-provisioner/main.go#L53

File: glusterblock/glusterblock-provisioner.go (hunk 2 above)
Comment (humblec) [inline glusterblock/glusterblock-provisioner.go]: Yes, imported and used.

File: glusterblock/glusterblock-provisioner.go (hunk 1 above)
Comment (humblec) [inline glusterblock/glusterblock-provisioner.go]: Done.

File: glusterblock/glusterblock-provisioner.go (hunk 3 above)
Comment (humblec) [inline glusterblock/glusterblock-provisioner.go]: Done.

**Comment by humblec** [comment]
@wongma7 Addressed the comments. Thanks..

**Comment by wongma7** [comment]
how to update dependencies:
* glide get k8s.io/apimachinery#c810e49 -v
* glide-vc --use-lock-file
OPTIONAL: 
* reset changes in glide.lock that you don't want/need then run glide install -v

**Comment by wongma7** [comment]
lgtm

**Pull request merged**